Source/WebSocket/Websocketpp/websocketpp_websocket.cpp

```diff
@@ -596,9 +596,22 @@ struct wspp_websocket_impl : public hc_websocket_impl, public std::enable_shared
             m_outgoingMessageQueue.pop();
         }
 
-        auto hr = XAsyncBegin(sendContext->message.async, shared_ptr_cache::store(sendContext), (void*)HCWebSocketSendMessageAsync, __FUNCTION__,
+        auto rawSendContext = shared_ptr_cache::store(sendContext);
+        if (rawSendContext == nullptr)
+        {
+            XAsyncComplete(sendContext->message.async, E_HC_NOT_INITIALISED, 0);
+            return E_HC_NOT_INITIALISED;
+        }
+
+        auto hr = XAsyncBegin(sendContext->message.async, rawSendContext, (void*)HCWebSocketSendMessageAsync, __FUNCTION__,
             [](XAsyncOp op, const XAsyncProviderData* data)
         {
+            auto httpSingleton = get_http_singleton(false);
+            if (nullptr == httpSingleton)
+            {
+                return E_HC_NOT_INITIALISED;
+            }
```

Comment:
if this condition is being met we are never removing the context form the shared_ptr_cache.